src/VisualStudio/IntegrationTest/New.IntegrationTests/InProcess/EditorVerifierInProcess.cs

```diff
@@ -130,6 +130,7 @@ public async Task CaretPositionAsync(int expectedCaretPosition, CancellationToke
                 [FeatureAttribute.Workspace, FeatureAttribute.SolutionCrawlerLegacy, FeatureAttribute.DiagnosticService, FeatureAttribute.ErrorSquiggles],
                 cancellationToken);
 
+            await Task.Delay(TimeSpan.FromSeconds(5));
```

Comment:
Seems suspicious - was this here because the editor doesn't provide an API for us to wait for diagnostics on their end?

If that is the case, we should likely have a 'WaitUntil function that waits until we have an expected value (with the normal cancellation token we use)

Reply:
fixing up.